credentials_update_test.go

```
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/taskcluster/taskcluster-client-go/tcclient"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+type RoutesTest struct {
+	Routes
+	t *testing.T
+}
+
+func TestCredentialsUpdate(t *testing.T) {
+	newCreds := Credentials{
+		ClientId:    "newClientId",
+		AccessToken: "newAccessToken",
+		Certificate: "newCertificate",
+	}
+
+	j, err := json.Marshal(&newCreds)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	body := string(j)
+	routes := NewRoutesTest(t)
+
+	response := routes.request("POST", body)
+	if response.Code != 405 {
+		t.Errorf("Should return 405, but returned %d", response.Code)
+	}
+
+	response = routes.request("PUT", "")
+	if response.Code != 400 {
+		t.Errorf("Should return 400, but returned %d", response.Code)
+	}
+
+	response = routes.request("PUT", body)
+	if response.Code != 200 {
+		content, _ := ioutil.ReadAll(response.Body)
```

nit: really we should report the error if it occurs